transactions/WFLY-17742_SupportTransactionsDuringGracefulShutdown

```diff
@@ -0,0 +1,146 @@
+
+===== Transactions
+
+*Current scenario.* The suspension logic developed in this subsystem is barely a call to the suspension hook of Narayana’s Recovery Coordinator, which only runs a single cycle to attempt the recovery of failed in-doubt transactions. As discussed previously, Narayana’s Recovery Coordinator does not necessarily recover all transactions before getting suspended as the “resume” action is supposed to happen (eventually). This will not necessarily be the case when WildFly is employed in cloud environments. Another important issue is that the current order (FIFO) used by the Suspend Controller during suspension results in a premature suspension of the Transactions subsystem. In other words, the recovery ability of this subsystem is suspended before dependent subsystems (e.g. EJB).
+*Proposed modifications:*
+* To address the shortcomings introduced with the FIFO order, a LIFO order should be used (discussed before)
+* The improvements to Narayana discussed above should be integrated in the Transactions subsystem
+  * In particular, the Transactions subsystem should become aware of how many in-doubt transactions there are in the Object Store and delay its suspension as long as all transactions are completed. In case (some of the) in-doubt transactions fail, the Transactions subsystem should wait as long as those transactions are resolved
+
+==== Note
+_Additional subsystems that depend on the Transactions subsystem should be added here, with details on how they should be handled._
+
+== Test Plan
+
+*wildfly-core.* As this proposal will not introduce new functionalities in wildfly-core, new testing is not needed.
+
+*Transactions and EJB subsystems.* Testing should be developed to make sure that in-doubt and in-flight transactions delay WildFly’s graceful shutdown. As a first step, we can test only WildFly on bare metal and then, if and when WildFly’s operator will be updated with modifications from this proposal, further testing might be developed in the cloud testing framework.
+
+== Community Documentation
+
+As WildFly’s graceful shutdown should be modified, WildFly’s documentation should reflect the different behaviour.
+
+== Release Note Content
+
+Graceful shutdown is modified to take into account cases where WildFly will not be restarted/resumed. In particular, to complete a graceful shutdown, all transactions must complete their lifecycles.
```

Comment:
I think I was referring to somehow an assumption I may have had (I can't recall sorry) that `In particular, to complete a graceful shutdown, all transactions must complete their lifecycles.` implies a limitation somehow but how I think it might be possible I just didn't parse it out right and it's not adding a limitation here just a draft of the release note text. Maybe in that case I could suggest to change `all transactions must complete their lifecycles` to `all transactions must now complete their lifecycles`?